.github/workflows/pr-labeler.yml

```diff
@@ -0,0 +1,85 @@
+# This action automatically labels Pull-Requests
+# based on files edited and no of lines changed.
+name: Size Labeler/Checker
+on:
+  pull_request_target:
+    types:
+      - opened
+      - synchronize
+      - edited
+jobs:
+  label-changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Add labels based on size
+        run: |
+          git fetch origin ${{ github.base_ref }}
+
+          # get all the changes made and changed files
+          CHANGES=$(git diff --ignore-all-space ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }})
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | awk -F'src/sage/' '{print $2}' | cut -d'/' -f1 | sed 's/\([^ ]\+\)/c: \1/g')
+
+          # ignore blank lines
+          CHANGES=$(echo "$CHANGES" | grep -vE '^[\+\-]\s*$')
+
+          # ignore non necessary lines from git diff
+          CHANGES=$(echo "$CHANGES" | grep -E '^[+\-]' | grep -vE '^\+\+\+|^\-\-\-')
+
+          # count total no of lines
+          CHANGES=$(echo "$CHANGES" | wc -l)
+
+          echo "CHANGES MADE: $CHANGES"
+
+          PR_NUMBER="${{ github.event.number }}"
+          API_URL="https://api.github.com/repos/${{ github.repository }}/issues/${PR_NUMBER}/labels"
+          AUTH_HEADER="Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"
+
+          DELETE_LABELS=("v: minimal" "v: small" "v: moderate" "v: large")
+
+          if [ "$CHANGES" -gt "$SMALL_THRESHOLD" ]; then
+            SIZE_LABEL="v: small"
```

Comment:
You missed that point from https://github.com/sagemath/sage/pull/37262#issuecomment-1992233557:

> I think you should use environment variables for repeatedly used literals (like v: minimal...).